src/integer/poly_over_z/from.rs

```diff
@@ -148,3 +224,46 @@ mod test_from_str {
         assert_eq!(PolyOverZ::from_str("4  1 2 3 -4").unwrap(), poly.unwrap());
     }
 }
+
+#[cfg(test)]
+mod test_from_poly_over_zq {
+    use crate::{integer::PolyOverZ, integer_mod_q::PolyOverZq};
+    use std::str::FromStr;
+
+    /// ensure that the conversion works with positive large entries
+    #[test]
+    fn large_positive() {
+        let poly = PolyOverZq::from_str(&format!("4  0 1 102 {} mod {}", u64::MAX - 58, u64::MAX))
+            .unwrap();
+
+        let poly_q = PolyOverZ::from(&poly);
+
+        let cmp_poly =
+            PolyOverZ::from_str(&format!("4  0 1 102 {} mod {}", u64::MAX - 58, u64::MAX)).unwrap();
```

Comment:
Why does this work?
Why is it possible to init a `PolyOverZ` with a string that contains `mod`?

Reply:
nice catch. I will open an issue